packages/core/src/html/codeblockProcessor.js

```diff
@@ -129,6 +129,26 @@ function highlightCodeBlock(node) {
   });
 }
 
+/**
+ * Adjust the class attribute of code blocks according to the global line numbers setting.
+ * Append the 'line-numbers' class if the global setting is true.
+ * @param node the code block element, which is the 'code' node
+ * @param showCodeLineNumbers true if line numbers should be shown, false otherwise
+ */
+function setCodeLineNumbers(node, showCodeLineNumbers) {
+  const existingClass = node.attribs.class || '';
+  const styleClassRegex = /(^|\s)(no-)?line-numbers($|\s)/;
+  const hasStyleClass = existingClass.match(styleClassRegex);
```

Comment:
one more tiny nit: you could also use the [`test`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test#description) method instead of `match`, its slightly faster / a little more semantic for js

(same for the one markdown-it/index.js)